independent-projects/extension-maven-plugin/src/main/java/io/quarkus/maven/ExtensionDescriptorMojo.java

```diff
@@ -678,17 +679,56 @@ public boolean visitLeave(DependencyNode node) {
     }
 
     private void addSource(ObjectNode extObject) throws MojoExecutionException {
-        Map<String, String> repo = ScmInfoProvider.getSourceRepo();
+        Scm scm = getScm();
+        String scmUrl = scm != null ? scm.getUrl() : null;
+
+        Map<String, String> repo = new ScmInfoProvider().getSourceRepo(scmUrl);
         if (repo != null) {
             ObjectNode metadata = getMetadataNode(extObject);
             for (Map.Entry<String, String> e : repo.entrySet()) {
-                // Tools may not be able to handle nesting in metadata, so do fake-nesting
-                metadata.put("scm-" + e.getKey(), e.getValue());
+                // Ignore if already set
+                String value = e.getValue();
+                String fieldName = "scm-" + e.getKey();
+                if (!metadata.has(fieldName) && value != null) {
+                    // Tools may not be able to handle nesting in metadata, so do fake-nesting
+                    metadata.put(fieldName, value);
+                }
 
             }
         }
     }
 
+    private Scm getScm() {
+        // We have three ways to do this; project.getScm() will query the derived model. Sadly, inherited <scm> entries are usually wrong, unless the parent is in the same project
+        // We can use getOriginalModel and getParent to walk the tree, but this will miss parents in poms outside the current execution
+        //We can also use more sophisticated techniques using repository managers, but we need to handle errors
```

Comment:
@holly-cummins missing the SCM from POMs outside the current repo is a good thing, IMO. It would mean it's a different project, so it would most probably not what we want to use.

Reply:
Or perhaps you mean only the reactor? Not actually outside the workspace?

Reply:
That's right, I was trying to say the second one, without using the word "reactor". I thought it would be clearer if I avoided maven jargon, but now I'm less sure. :)